### DISCUSSION TITLE
Jaeger use zincsearch as storage backend

### DISCUSSION THREAD

**Discussion by dangweiwu** [zinclabs/zinc#379] [General]
jaeger use Elasticsearch as storage backend，i use zincsearch instead of Elasticsearch，but，jaeger not working，jaeger request “HEAD /”  but response 404。could add this api ？

**Comment by prabhatsharma** [comment]
We don't have HEAD for any API implemented. We will need to check it. Do you have the details of API which is being called?

**Comment by dangweiwu** [comment]
I try to zincsearch instead of elasticsearch with jaeger storage backend，at last，jaeger not working，jaeger-collector and jaeger-query request zincsearch but failed with 400。I found zincSearch could not replace elasticsearch。Do you have any plans to  adapt zincsearch to gaeger。

**Comment by prabhatsharma** [comment]
We do not have a list of applications like Jaeger that we like to support migration from ES. If you want support for an application you can help us by providing the APIs that need to be supported for that application and we can think to implement them.

**Comment by morph027** [comment]
Looks like this commit is adding some support if you use /es/ as endpoint: https://github.com/zinclabs/zinc/commit/5c0318f52cc09792955353ee028cf8e264676407

**Comment by prabhatsharma** [comment]
/es has been there for quite a while and we support a lot of ES APIs. If that is all needed then it should work fine.

**Comment by morph027** [comment]
Jeah, sure, was just looking for the HEAD calls within code. I'm trying to figure out if i can use zinc for gitea as indexer, which makes some similar requests. Got it working beyond the first HEAD by building latest binary from main branch. I'm now stuck at HEAD for /_alias...

**Reply by prabhatsharma** [reply]
Can you provide the API call reference on what the endpoint is and what it expects and if it's easy we should be able to implement it fast.

**Comment by morph027** [comment]
Jip, still trying to figure out by reading gitea code and/or adding a mitm proxy. However, this is the logs from zinc:

```
[GIN-debug] redirecting request 307: /es/ --> /es/
[GIN-debug] redirecting request 307: /es/ --> /es/
{"level":"info","method":"HEAD","code":200,"took":0,"time":"2022-08-04T19:07:26+02:00","message":"/es/"}
{"level":"info","method":"HEAD","code":200,"took":0,"time":"2022-08-04T19:07:26+02:00","message":"/es/"}
[GIN-debug] redirecting request 307: /es/ --> /es/
[GIN-debug] redirecting request 307: /es/ --> /es/
{"level":"info","method":"HEAD","code":200,"took":0,"time":"2022-08-04T19:07:26+02:00","message":"/es/"}
{"level":"info","method":"HEAD","code":200,"took":0,"time":"2022-08-04T19:07:26+02:00","message":"/es/"}
{"level":"info","method":"HEAD","code":200,"took":0,"time":"2022-08-04T19:07:26+02:00","message":"/es/gitea_codes.v1"}
{"level":"info","method":"HEAD","code":200,"took":0,"time":"2022-08-04T19:07:26+02:00","message":"/es/gitea_issues"}
{"level":"error","method":"GET","code":404,"took":0,"time":"2022-08-04T19:07:26+02:00","message":"/es/_alias"}
{"level":"info","method":"GET","code":404,"took":0,"time":"2022-08-04T19:07:26+02:00","message":"/es/_alias"}
```

btw, thanks for your quick help, much appreciated :heart: Not sure if we should split the gitea topic from jaeger.

**Reply by morph027** [reply]
This is what an ES is returning:

```
curl elasticsearch:9200/_alias
{"gitea_issues":{"aliases":{}},"gitea_codes.v1":{"aliases":{"gitea_codes":{}}}}
```

**Reply by morph027** [reply]
But most certainly due to https://github.com/zinclabs/zinc/issues/141 ;)

**Reply by prabhatsharma** [reply]
Yeah, we don't have support for alias for now.
